### ISSUE TITLE
Test coverage does not include react component for selective jsdom env

### ISSUE THREAD

**Issue by kwonoj** [wallabyjs/public#2465]
### Issue description or question

Repro: https://github.com/kwonoj/wallaby-test

Jest configuration's set to node.js by default, and there are some react component test cases using jsdom env via optin using top-level comment. Wallaby's autoconfig runs test cases successfully, but test explorer & coverage doesn't include actual component files.

// for non-react component test:
![image](https://user-images.githubusercontent.com/1210596/86039952-11ce9280-b9f8-11ea-9d93-c295d9e0f3ae.png)

`index.ts` correctly displays test coverage

// for react component test
![image](https://user-images.githubusercontent.com/1210596/86040018-2b6fda00-b9f8-11ea-9be8-8f3062bf468b.png)

also test explorer does not include component files

![image](https://user-images.githubusercontent.com/1210596/86040081-404c6d80-b9f8-11ea-86aa-d282a4ffc054.png)

### Wallaby diagnostics report

```
{
  editorVersion: '1.46.1',
  pluginVersion: '1.0.212',
  editorType: 'VSCode',
  osVersion: 'linux 5.4.0-7634-generic',
  nodeVersion: 'v14.1.0',
  coreVersion: '1.0.920',
  config: {
    trace: true,
    workers: { initial: 2, regular: 2, recycle: false },
    diagnostics: {
      jest: {
        config: {
          configs: [
            {
              automock: false,
              cache: true,
              cacheDirectory: '/tmp/jest_rs',
              clearMocks: false,
              coveragePathIgnorePatterns: [ '<homeDir>/github/wallaby-test/node_modules/', '<homeDir>/github/wallaby-test/spec/.*\\.(ts|js)$', '<homeDir>/github/wallaby-test/build/.*\\.(ts|js)$' ],
              cwd: '<homeDir>/github/wallaby-test',
              dependencyExtractor: undefined,
              detectLeaks: undefined,
              detectOpenHandles: undefined,
              displayName: undefined,
              errorOnDeprecated: false,
              extraGlobals: [],
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: { 'ts-jest': { tsConfig: 'spec/tsconfig.json', diagnostics: false, babelConfig: { comments: false } }, __JEST_DEV__: true },
              haste: { computeSha1: false, throwOnModuleCollision: false },
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [ 'js', 'jsx', 'json', 'ts', 'tsx' ],
              moduleLoader: undefined,
              moduleNameMapper: [],
              modulePathIgnorePatterns: [ '/.tmp/' ],
              modulePaths: undefined,
              name: '9a23efbe40036cfd06832968c9fb0d60',
              prettierPath: 'prettier',
              resetMocks: false,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: '<homeDir>/github/wallaby-test',
              roots: [ '<homeDir>/github/wallaby-test' ],
              runner: 'jest-runner',
              setupFiles: [],
              setupFilesAfterEnv: [],
              skipFilter: false,
              skipNodeResolution: undefined,
              snapshotResolver: undefined,
              snapshotSerializers: [],
              testEnvironment: '<homeDir>/github/wallaby-test/node_modules/jest-environment-jsdom/build/index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [],
              testPathIgnorePatterns: [ '/.tmp/', '/spec/mocha', '/spec/e2e' ],
              testRegex: [ '(-spec)\\.(ts|tsx)$' ],
              testRunner: '<homeDir>/github/wallaby-test/node_modules/jest-circus/runner.js',
              testURL: 'http://localhost',
              timers: 'real',
              transform: [
                [ '^.+\\.jsx?$', '<homeDir>/github/wallaby-test/node_modules/babel-jest/build/index.js', {} ],
                [ '^.+\\.tsx?$', '<homeDir>/github/wallaby-test/node_modules/ts-jest/dist/index.js', {} ]
              ],
              transformIgnorePatterns: [ '/node_modules/' ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            collectCoverage: false,
            collectCoverageFrom: [ 'src/**/*.{ts, tsx}', '!**/*.d.ts' ],
            collectCoverageOnlyFrom: undefined,
            coverageDirectory: '<homeDir>/github/wallaby-test/coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json-summary' ],
            coverageThreshold: undefined,
            detectLeaks: undefined,
            detectOpenHandles: undefined,
            enabledTestsMap: undefined,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: undefined,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 11,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            outputFile: undefined,
            passWithNoTests: undefined,
            projects: [],
            replname: undefined,
            reporters: undefined,
            rootDir: '<homeDir>/github/wallaby-test',
            runTestsByPath: false,
            silent: undefined,
            skipFilter: false,
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: '<homeDir>/github/wallaby-test/node_modules/@jest/test-sequencer/build/index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: undefined,
            watch: false,
            watchAll: false,
            watchPlugins: undefined,
            watchman: true
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [],
            watchPathIgnorePatterns: [ '/node_modules/', '\\./dist/|\\./build/|\\./coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/', '/.tmp/', '/spec/mocha', '/spec/e2e' ],
            testPathIgnorePatterns: [ '/node_modules/', '\\./dist/|\\./build/|\\./coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/', '/.tmp/', '/spec/mocha', '/spec/e2e' ],
            testMatch: [],
            testRegex: [ '(-spec)\\.(ts|tsx)$' ]
          }
        }
      }
    },
    filesWithCoverageCalculated: [ 'src/**/*.{ts, tsx}', '!**/*.d.ts' ],
    globalSetup: false,
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    micromatch: true,
    files: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true },
      {
        pattern: '\\./dist/|\\./build/|\\./coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /\.\/dist\/|\.\/build\/|\.\/coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true
      },
      { pattern: '/.tmp/', regexp: /\/.tmp\//, ignore: true, trigger: true, load: true },
      { pattern: '/spec/mocha', regexp: /\/spec\/mocha/, ignore: true, trigger: true, load: true },
      { pattern: '/spec/e2e', regexp: /\/spec\/e2e/, ignore: true, trigger: true, load: true },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '(-spec)\\.(ts|tsx)$', regexp: /(-spec)\.(ts|tsx)$/, ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '/node_modules/', regexp: /\/node_modules\//, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: '\\./dist/|\\./build/|\\./coverage/|/\\.cache/|/\\.idea/|/\\.vscode/|/\\.git/|/\\.gitlab/',
        regexp: /\.\/dist\/|\.\/build\/|\.\/coverage\/|\/\.cache\/|\/\.idea\/|\/\.vscode\/|\/\.git\/|\/\.gitlab\//,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: '/.tmp/', regexp: /\/.tmp\//, ignore: true, trigger: true, load: true, test: true },
      { pattern: '/spec/mocha', regexp: /\/spec\/mocha/, ignore: true, trigger: true, load: true, test: true },
      { pattern: '/spec/e2e', regexp: /\/spec\/e2e/, ignore: true, trigger: true, load: true, test: true },
      { pattern: '(-spec)\\.(ts|tsx)$', regexp: /(-spec)\.(ts|tsx)$/, ignore: false, trigger: true, load: true, test: true, order: 2 }
    ],
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    extensions: {},
    env: {
      type: 'node',
      params: {},
      runner: '<homeDir>/.fnm/node-versions/v14.1.0/installation/bin/node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'module.exports = () => ({\n  autoDetect: true,\n  trace: true,\n  workers: {\n    initial: 2,\n    regular: 2\n  }\n});\n'
  },
  packageJSON: {
    dependencies: { react: '^16.13.1', 'react-dom': '^16.13.1', 'react-redux': '^7.2.0', redux: '^4.0.5' },
    devDependencies: { '@testing-library/react': '^10.4.3', '@types/jest': '^26.0.3', '@types/react': '^16.9.41', jest: '^26.1.0', 'jest-circus': '^26.1.0', 'ts-jest': '^26.1.1', typescript: '^3.9.5' }
  },
  fs: { numberOfFiles: 13 },
  debug: [
    '2020-06-29T18:04:45.389Z angular/cli config Angular CLI not found.\n',
    '2020-06-29T18:04:45.464Z jest/config Detected Jest.\n',
    '2020-06-29T18:04:45.464Z jest/config Configured Jest.\n',
    '2020-06-29T18:04:45.465Z project Wallaby Node version: v14.1.0\n',
    '2020-06-29T18:04:45.465Z project Wallaby config: <homeDir>/github/wallaby-test/wallaby.js\n',
    '2020-06-29T18:04:46.134Z project File cache: <homeDir>/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.212/projects/7749b99ff2f3a314\n',
    '2020-06-29T18:04:46.141Z uiService Listening port 51235\n',
    '2020-06-29T18:04:46.145Z workers Parallelism for initial run: 2, for regular run: 2\n',
    '2020-06-29T18:04:46.145Z workers Starting run worker instance #0\n',
    '2020-06-29T18:04:46.145Z workers Starting run worker instance #1\n',
    '2020-06-29T18:04:46.146Z workers Web server is listening at 40763\n',
    '2020-06-29T18:04:46.146Z project File cache requires some updates, waiting required files from IDE\n',
    '2020-06-29T18:04:46.160Z project Stopping process pool\n',
    '2020-06-29T18:04:46.161Z project Test run started; run priority: 3\n',
    '2020-06-29T18:04:46.162Z project Running all tests\n',
    '2020-06-29T18:04:46.163Z workers Starting test run, priority: 3\n',
    '2020-06-29T18:04:46.163Z workers Distributing tests between 2 workers\n',
    '2020-06-29T18:04:46.164Z workers Running tests in parallel\n',
    '2020-06-29T18:04:46.164Z nodeRunner Starting sandbox [worker #0, session #lf0yb]\n',
    '2020-06-29T18:04:46.164Z nodeRunner Starting sandbox [worker #1, session #db9d6]\n',
    '2020-06-29T18:04:46.164Z nodeRunner Preparing sandbox [worker #0, session #lf0yb]\n',
    '2020-06-29T18:04:46.164Z nodeRunner Preparing sandbox [worker #1, session #db9d6]\n',
    '2020-06-29T18:04:46.247Z workers Started run worker instance (delayed) #0\n',
    '2020-06-29T18:04:46.247Z nodeRunner Prepared sandbox [worker #0, session #lf0yb]\n',
    '2020-06-29T18:04:46.247Z workers [worker #0, session #lf0yb] Running tests in sandbox\n',
    '2020-06-29T18:04:46.254Z workers Started run worker instance (delayed) #1\n',
    '2020-06-29T18:04:46.254Z nodeRunner Prepared sandbox [worker #1, session #db9d6]\n',
    '2020-06-29T18:04:46.254Z workers [worker #1, session #db9d6] Running tests in sandbox\n',
    '2020-06-29T18:04:50.016Z workers [lf0yb] Loaded unknown number of test(s)\n',
    '2020-06-29T18:04:50.017Z workers [lf0yb] Test executed: works\n',
    '2020-06-29T18:04:50.020Z workers [lf0yb] Run 1 test(s), skipped 0 test(s)\n',
    '2020-06-29T18:04:50.022Z workers [lf0yb] Sandbox is responsive, closing it\n',
    '2020-06-29T18:04:50.586Z workers [db9d6] Loaded unknown number of test(s)\n',
    '2020-06-29T18:04:50.586Z workers [db9d6] Test executed: works\n',
    '2020-06-29T18:04:50.588Z workers [db9d6] Run 1 test(s), skipped 0 test(s)\n',
    '2020-06-29T18:04:50.590Z workers [db9d6] Sandbox is responsive, closing it\n',
    '2020-06-29T18:04:50.590Z workers Merging parallel test run results\n',
    '2020-06-29T18:04:50.591Z project Test run finished\n',
    '2020-06-29T18:04:50.591Z project Processed console.log entries\n',
    '2020-06-29T18:04:50.592Z project Processed loading sequences\n',
    '2020-06-29T18:04:50.592Z project Processed executed tests\n',
    '2020-06-29T18:04:50.593Z project Processed code coverage\n',
    '2020-06-29T18:04:50.603Z project Test run result processed and sent to IDE\n',
    '2020-06-29T18:04:52.280Z uiService UI client connected\n',
    '2020-06-29T18:04:52.281Z uiService Outgoing message ui:handshake\n',
    '2020-06-29T18:04:52.289Z uiService Incoming message ui:start\n',
    '2020-06-29T18:04:52.289Z uiService Outgoing message ui:summary\n',
    '2020-06-29T18:04:52.290Z uiService Outgoing message ui:files\n'
  ]
}
```

**Comment by smcenlly** [comment]
Thanks for providing the sample repo. The reason that this isn't working for you is that there is a problem in your `collectCoverageFrom`  **jest.config.js** setting. You will have the same problem if you use the jest cli with `npx jest --collectCoverage`.

There was also another issue causing Wallaby's live comments to not work because ts-jest's babel processor is not including comments in its output.

To fix these problems, please update your jest configuration as shown below:

**jest.config.js**
```diff
const isWallaby = !!process.env.WALLABY_ENV;

module.exports = {
  preset: "ts-jest/presets/js-with-babel",
  testRegex: "(-spec)\\.(ts|tsx)$",
  testRunner: "jest-circus/runner",
  testMatch: null,
  bail: false,
  resetMocks: false,
  resetModules: false,
  globals: {
    "ts-jest": {
      tsConfig: "spec/tsconfig.json",
      diagnostics: false,
      babelConfig: {
-       comments: false,
+       comments: isWallaby,
      },
    },
    __JEST_DEV__: true,
  },
  moduleFileExtensions: [ "js", "jsx", "json", "ts", "tsx" ],
  testPathIgnorePatterns: [ "/.tmp/", "/spec/mocha", "/spec/e2e" ],
  modulePathIgnorePatterns: [ "/.tmp/" ],
  coverageReporters: !isWallaby ? [ "json-summary" ] : undefined,
- collectCoverageFrom: [ "src/**/*.{ts, tsx}", "!**/*.d.ts" ],
+ collectCoverageFrom: [ "src/**/*.[t]s?(x)", "!**/*.d.ts" ],
  coveragePathIgnorePatterns: [
    "<rootDir>/node_modules/",
    "<rootDir>/spec/.*\\.(ts|js)$",
    "<rootDir>/build/.*\\.(ts|js)$",
  ]
};
```

**Comment by kwonoj** [comment]
Appreciate for the quick help!
